django/tasks/base.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import unicode_literals
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.utils.module_loading import import_string
+
+from .exceptions import InvalidTaskBackendError
+
+DEFAULT_TASK_ALIAS = 'default'
+
+# backends cache to make sure we don't create one per connection
+# all the backends should be thread-safe
+backends = {}
+
+
+def _create_task_backend(alias):
+    try:
+        conf = settings.QUEUES[alias]
+    except KeyError:
+        if alias != DEFAULT_TASK_ALIAS:
```

Comment:
The idea behind that is that the admin for instance can send to the "email"
queue and the user can just use one.... The same exists for instance for
the template fragment cache which uses xxx if defined or falls back to
default

On Thu, Jun 4, 2015, 19:03 Carl Meyer <notifications@github.com> wrote:

> In django/tasks/base.py
> <https://github.com/django/django/pull/4763#discussion_r31747347>:
>
> > +from django.utils.module_loading import import_string
> > +
> > +from .exceptions import InvalidTaskBackendError
> > +
> > +DEFAULT_TASK_ALIAS = 'default'
> > +
> > +# backends cache to make sure we don't create one per connection
> > +# all the backends should be thread-safe
> > +backends = {}
> > +
> > +
> > +def _create_task_backend(alias):
> > +    try:
> > +        conf = settings.QUEUES[alias]
> > +    except KeyError:
> > +        if alias != DEFAULT_TASK_ALIAS:
>
> (In case it's not clear, that seems to me like a really bad idea and a
> recipe for bugs. Typos should not be silently hidden.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/django/django/pull/4763/files#r31747347>.
>